### PR TITLE
feat: add transaction table

### PR DESCRIPTION
**Pull request by tribiec**
PR in draft mode (building PR)

## Demo

https://github.com/TransbankDevelopers/transbank-plugin-prestashop-webpay-rest/assets/16856300/19cbf5f9-833f-438f-add8-480db88c364c

image

### CODE REVIEW

**Comment by mastudillot** [comment]
> left some comments. I think it would be good to include a field for the payment type (Oneclick, Webpay Plus).

Added ✅

File: webpay/src/Grid/TransactionsStatusChoiceProvider.php [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+use PrestaShop\PrestaShop\Core\Form\FormChoiceProviderInterface;
+use PrestaShop\Module\WebpayPlus\Model\TransbankWebpayRestTransaction;
+
+class TransactionsStatusChoiceProvider implements FormChoiceProviderInterface
+{
+    public function getChoices()
+    {
+        return [
+            'Inicializado' => TransbankWebpayRestTransaction::STATUS_INITIALIZED,
+            'Failed' => TransbankWebpayRestTransaction::STATUS_FAILED,
```
[the remark this answers is not in the document]
Comment (mastudillot) [reply webpay/src/Grid/TransactionsStatusChoiceProvider.php:14]: Changed

File: webpay/src/Grid/TransactionsGridDefinitionFactory.php [hunk 2]
```diff
@@ -0,0 +1,229 @@
...
+    }
+
+    protected function getFilters(): FilterCollectionInterface
+    {
+        return (new FilterCollection())->add(
+            (new Filter('cart_id', TextType::class))
+                ->setAssociatedColumn('cart_id')
+                ->setTypeOptions([
+                    'attr' => [
+                        'placeholder' => 'Buscar por Id de carrito'
```
[the remark this answers is not in the document]
Comment (mastudillot) [reply webpay/src/Grid/TransactionsGridDefinitionFactory.php:114]: Removed

File: webpay/src/Grid/TransactionsGridDefinitionFactory.php [hunk 3]
```diff
@@ -0,0 +1,229 @@
...
+                    ])
+            )
+            ->add((new DataColumn('order_id'))
+                    ->setName('Id del pedido')
+                    ->setOptions([
+                        'field' => 'order_id',
+                    ])
+            )
+            ->add((new DataColumn('response_code'))
+                    ->setName('C. Resp')
```
[the remark this answers is not in the document]
Comment (mastudillot) [reply webpay/src/Grid/TransactionsGridDefinitionFactory.php:57]: changed

File: webpay/config/services.yml [hunk 4]
```diff
@@ -10,87 +10,114 @@ services:
...
+    parent: "prestashop.core.grid.abstract_query_builder"
+    arguments:
+      - "@=service('prestashop.adapter.legacy.context').getContext().language.id"
+      - "@=service('prestashop.adapter.legacy.context').getContext().shop.id"
+    public: true
+
+  webpay.grid.data.factory.transactions_data_factory:
+    class: 'PrestaShop\PrestaShop\Core\Grid\Data\Factory\DoctrineGridDataFactory'
+    arguments:
+      - "@webpay.grid.query.transactions_query_builder" # service id of our query builder
```
[the remark this answers is not in the document]
Comment (mastudillot) [reply webpay/config/services.yml:112]: Removed

**Review by victormendoza96 — APPROVED** [review]

**Review by Matiasnickolas — APPROVED** [review]
LGTM 🤙

**Pull request merged**